tests/test_slots.py

```diff
@@ -300,11 +302,28 @@ def test_advance_physical_slots(self):
                           [self.me, self.other, self.leadermem], None, SyncState.empty(), None, None)
         global_config.update(cluster)
         self.s.sync_replication_slots(cluster, self.tags)
-        with patch.object(SlotsHandler, '_query', Mock(side_effect=[[('blabla', 'physical', 12345, None, None, None,
-                                                                      None, None)], Exception])) as mock_query, \
+        with patch.object(SlotsHandler, '_query', Mock(side_effect=[[('blabla', 'physical', None, 12345, None, None,
+                                                                      None, None, None)], Exception])) as mock_query, \
                 patch('patroni.postgresql.slots.logger.error') as mock_error:
             self.s.sync_replication_slots(cluster, self.tags)
             self.assertEqual(mock_query.call_args[0],
                              ("SELECT pg_catalog.pg_replication_slot_advance(%s, %s)", "blabla", '0/303A'))
             self.assertEqual(mock_error.call_args[0][0],
                              "Error while advancing replication slot %s to position '%s': %r")
+
+        with patch.object(SlotsHandler, '_query', Mock(side_effect=[[('test_1', 'physical', 1, 12345, None, None,
+                                                                      None, None, None)], Exception])), \
+                patch.object(SlotsHandler, 'drop_replication_slot', Mock(return_value=(False, True))):
+            self.s.sync_replication_slots(cluster, self.tags)
+
+        with patch.object(SlotsHandler, '_query', Mock(side_effect=[[('test_1', 'physical', 1, 12345, None, None,
+                                                                      None, None, None)], Exception])), \
+                patch.object(SlotsHandler, 'drop_replication_slot', Mock(return_value=(True, False))), \
+                patch('patroni.postgresql.slots.logger.warning') as mock_warning:
+            self.s.sync_replication_slots(cluster, self.tags)
+            self.assertEqual(mock_warning.call_args_list[-1][0], ("Unable to drop replication slot '%s', slot is active", 'test_1'))
```

Comment:
But it is impossible situation. We can't get the error but not get a warning.

Reply:
Maybe I just misplaced the comment.

I was just wondering if we want to assert somewhere in the unit tests that the message was issued by Patroni when dropping a slot because of that reason.

Reply:
I see. Yeah, actually I want to do it, but there are a lot of other error messages, and the order of them is not deterministic, therefore the right one needs to be search in the call_args_list and I am too lazy to do it :)